### ISSUE TITLE
map is not sorted by key

### ISSUE THREAD

**Issue by miworking3** [apple/swift-protobuf#1075]
* what OS you are developing on (Linux or macOS, including the version)
macOS
* for macOS, what version of Xcode you are using (`xcodebuild -version`),
  for Linux, what version of Swift (`swift --version`)
11.7
* what version of Swift is your code set to compile with (i.e. from project
  settings, etc.)
5.2.4

* what branch/tag of SwiftProtobuf you are using (1.0.0, etc.)
1.12.0 Release

* Problem Description
 the `map` is not implemented as defined by [proto3](https://developers.google.com/protocol-buffers/docs/proto):
```When generating text format for a .proto, maps are sorted by key.```
while the output of `textFormatString()` of a dictionary in SWIFT is randomly sorted.

* Expected Result
If swift-protobuf decides to use `Dictionary` as the backed up data structure for `map` in proto3,  the result of `dictionary.textFormatString()` is expected to be sorted by the key.
e.g.,
```
var map = [String:String]()
map["b"] = "bbb"
map["a"] = "aaa"

print(map) // ["b": "bbb", "a": "aaa"]
print(map.textFormatString()) // [ "a": "aaa", "b": "bbb"]
```
* Proposed fix:
A fix for this line in [TextFormatEncodingVisitor.swift](https://github.com/apple/swift-protobuf/blob/master/Sources/SwiftProtobuf/TextFormatEncodingVisitor.swift#L616)

before
```
for (k,v) in map {
```
after:
```
for (k,v) in map.sorted { $0.0  < $1.0 }.map { $0 } {
       
```

**Comment by tbkka** [comment]
We would appreciate a Pull Request with this change.  Please also include a test case.  There should be some existing tests for map encoding/decoding in TextFormate that could be easily updated.

**Comment by thomasvl** [comment]
There's even a conformance test, but the data set must always pass for Swift.

https://github.com/protocolbuffers/protobuf/blob/e667bf6eaaa2fb1ba2987c6538df81f88500d030/conformance/text_format_conformance_suite.cc#L317-L375

**Comment by thomasvl** [comment]
fyi - I reached out to the protobuf team, it looks like that conformance test parses the TextFormat and compares the binary forms, so it doesn't actually check the raw TextFormat.

**Comment by thomasvl** [comment]
Also, I don't think that proposed fix is drop in.  Since a `Dictionary`'s keys are only `Hashable` it doesn't mean they are `Comparable`, so odds are a few things might need tweaking to get it to work.

**Comment by thomasvl** [comment]
From a quick peek:
- `MapKeyType` could require `Comparable` on `BaseType`, but `Bool` doesn't conform.  Adding a conformance seems like a _no-no_ since we'd have to make it public so it would bleed into everything that use SwiftProtobuf
- We might be able to extend `Visitor` with new methods that call out the `Comparable` requirements and add special cases for the `Bool` keys also, and then have default impls that direct things back to a common impl for the `Visitor`s that don't care, but that extra indirection seems less than idea.

Nothing else jumps at me as idea for how to add this easily.

**Comment by thomasvl** [comment]
Fixed via #1076
